src/include/daos/mem.h

```diff
@@ -355,8 +370,22 @@ umem_cache_off2ptr(struct umem_store *store, umem_off_t offset);
  *
  * \return	MD-blob offset
  */
-umem_off_t
-umem_cache_ptr2off(struct umem_store *store, const void *ptr);
+static inline umem_off_t
+umem_cache_ptr2off(struct umem_store *store, const void *ptr)
+{
+	struct umem_cache *cache = store->cache;
+	unsigned int       idx;
+	umem_off_t         base_off;
+
+	idx = (ptr - cache->ca_base) >> cache->ca_page_shift;
+	D_ASSERTF((idx <= cache->ca_mem_pages), "ptr = %p, id = %u", ptr, idx);
+
+	base_off = cache->ptr2off[idx];
+	D_ASSERTF((base_off != (-1UL)), "base_off = %lu, ca_base = %p, cache->ca_mem_pages = %d",
+		  base_off, cache->ca_base, cache->ca_mem_pages);
+
+	return base_off + ((ptr - cache->ca_base) & cache->ca_page_mask);
+}
```

Comment:
Besides making these two functions as inline function, it also avoided accessing umem_page_info, instead it accesses ptr2off/off2ptr array which is likely staying in CPU cache?